### DISCUSSION TITLE
Run a prompt over a selected piece of code (added to the context)

### DISCUSSION THREAD

**Discussion by a4w** [cline/cline#473] [Feature Requests]
Run a prompt over a selected piece of code (added to the context). Similar to the UX of "Continue" from the VS code marketplace

**Comment by PhunkyBob** [comment]
The feature I would like: 
- I select a part of the code (eg. a function)
- I prompt something (eg. "add a xyz parameter that ...")
- Cline only edit the part I've selected.

and
- I position my cursor somewhere in the code
- I prompt something to ADD at the cursor, without changing anything else (eg. "insert a sub title tight here")

**Comment by PhunkyBob** [comment]
I've created a PR for this request: https://github.com/cline/cline/pull/631
